### PR TITLE
run conf: fix double gossip host when set in config toml

### PR DESCRIPTION
**Pull request by SOELTH**
When manually specifying gossip host, run conf erroneously sets it twice. Fixes run conf to only specify the default if an alternative isn't specified in config

### CODE REVIEW

File: src/app/fdctl/run/run.c [hunk 1]
```diff
@@ -311,6 +307,11 @@ solana_labs_main( void * args ) {
   ADDH( "--gossip-port", config->gossip.port );
   if( strcmp( config->gossip.host, "" ) )
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/run.c:308]:
```suggestion
  if( strcmp( config->gossip.host, "" ) ) {
```

File: src/app/fdctl/run/run.c [hunk 2]
```diff
@@ -311,6 +307,11 @@ solana_labs_main( void * args ) {
...
     ADD( "--gossip-host", config->gossip.host );
+  else{
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/run.c:310]:
```suggestion
  } else {
```

**Review by mmcgee-jump — DISMISSED** [review]
Thank you! Minor style nits.

**Comment by mmcgee-jump** [comment]
Sorry for the nits, could you please squash commits as well? Thanks for the contribution! (The GH UI is terrible when accepting suggestions)

**Comment by SOELTH** [comment]
> Sorry for the nits, could you please squash commits as well? Thanks for the contribution! (The GH UI is terrible when accepting suggestions)

No worries! Should be good now! Agree the GH UI is terrible for this 🫠

**Review by mmcgee-jump — APPROVED** [review]

**Pull request merged**